addons/controllers/kapp-controller/kappcontrollerconfig_controller.go

```diff
@@ -57,28 +58,24 @@ func (r *KappControllerConfigReconciler) Reconcile(ctx context.Context, req ctrl
 		r.Log.Error(err, "Unable to fetch kappControllerConfig resource")
 		return ctrl.Result{}, err
 	}
-	// Deepcopy to prevent client-go cache conflict
-	kappControllerConfig = kappControllerConfig.DeepCopy()
 
-	// get the parent cluster name from owner reference
-	// if the owner reference doesn't exist, use the same name as config CR
-	clusterNamespacedName := req.NamespacedName
-	cluster := &clusterapiv1beta1.Cluster{}
-	for _, owner := range kappControllerConfig.OwnerReferences {
-		if owner.Kind == constants.ClusterKind {
-			clusterNamespacedName.Name = owner.Name
-			break
-		}
+	annotations := kappControllerConfig.GetAnnotations()
+	if _, ok := annotations[constants.TKGAnnotationTemplateConfig]; ok {
+		r.Log.Info(fmt.Sprintf("resource '%v' is a config template. Skipping reconciling", req.NamespacedName))
+		return ctrl.Result{}, nil
 	}
 
-	// verify that the cluster related to config is present
-	if err := r.Client.Get(ctx, clusterNamespacedName, cluster); err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info(fmt.Sprintf("Cluster %s/%s not found", clusterNamespacedName.Namespace, clusterNamespacedName.Name))
+	// Deepcopy to prevent client-go cache conflict
+	kappControllerConfig = kappControllerConfig.DeepCopy()
+
+	cluster, err := cutil.GetOwnerCluster(ctx, r.Client, kappControllerConfig, req.Namespace, constants.KappControllerDefaultRefName)
+	if err != nil {
+		if apierrors.IsNotFound(err) && cluster != nil {
+			r.Log.Info(fmt.Sprintf("'%s/%s' is listed as owner reference but could not be found",
+				cluster.Namespace, cluster.Name))
 			return ctrl.Result{}, nil
 		}
-
-		log.Error(err, fmt.Sprintf("unable to fetch cluster %s/%s", clusterNamespacedName.Namespace, clusterNamespacedName.Name))
+		r.Log.Info("could not determine owner cluster")
```

Comment:
Please log the error message as well so its clear what error is

Reply:
log as error now with error message